.github/workflows/cygwin.yml

```
@@ -25,15 +25,16 @@ jobs:
           submodules: recursive
           fetch-depth: 0
       - name: Install Cygwin
-        uses: cygwin/cygwin-install-action@006ad0b0946ca6d0a3ea2d4437677fa767392401 # v4
+        uses: egor-tensin/setup-cygwin@d2c752bab416d4b0662591bd366fc2686297c82d   #v4
         with:
           platform: x86_64
           install-dir: 'C:\tools\cygwin'
           packages: >-
-            python39-devel python39-pip python-pip-wheel python-setuptools-wheel
-            liblapack-devel liblapack0 gcc-fortran gcc-g++ git dash cmake ninja
+            python39-devel=3.9.16-1 python39-pip python-pip-wheel
```

It looks like even though `python39-devel=3.9.16-1` is specified, the `python39` package itself is still installed at 3.9.18-1, which I am guessing is why `pip install` [is stalling](https://github.com/numpy/numpy/actions/runs/7695645129/job/20968978747?pr=25716#step:8:44) at the download for `pytest_xdist`. I suspect that also pinning `python39=3.9.16-1` would fix this.

```suggestion
            python39=3.9.16-1 python39-devel=3.9.16-1 python39-pip python-pip-wheel
```

---

Thanks